content/v2.0/introduction/getting-started.md

```diff
@@ -138,29 +162,1261 @@ Visit your application in the browser at [http://localhost:2300](http://localhos
+
+```ruby
+# app/actions/books/index.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  module Actions
+    module Books
+      class Index < Bookshelf::Action
+        def handle(*, response)
+          books = [
+            {title: "Test Driven Development"},
+            {title: "Patterns of Enterprise Application Architecture"}
+          ]
+
+          response.format = :json
+          response.body = books.to_json
+        end
+      end
+    end
+  end
+end
+```
+
+If we run our spec, it now passes!
+
+```shell
+bundle exec rspec spec/requests/books/index_spec.rb
+
+Randomized with seed 16566
+
+GET /books
+  returns a list of books
+
+Finished in 0.02378 seconds (files took 0.49411 seconds to load)
+1 example, 0 failures
+```
+
+## Persisting books
+
+Of course, returning a static list of books is not particularly useful.
+
+Let's address this by retrieving books from a database.
+
+<p class="notice">
+  Integrated support for peristence based on <a href="https://rom-rb.org/">rom-rb</a> is coming in Hanami's 2.1 release. For now, we can bring our own simple rom-rb configuration to allow us to store books in a database.
+</p>
+
+### Adding persistence using rom-rb
+
+Let's add just enough rom-rb to get persistence working using Postgres.
+
+First, add these gems to the Gemfile and run `bundle install`:
+
+```ruby
+# Gemfile
+gem "rom"
+gem "rom-sql"
+gem "pg"
+
+group :test do
+  gem "database_cleaner-sequel"
+end
+```
+
+<p class="notice">
+  If you do not have Postgres installed, you can install it using <a href="https://brew.sh/">Homebrew</a>,  <a href="https://asdf-vm.com/">asdf</a> or by following the installation instruction on the <a href="https://www.postgresql.org/">PostgreSQL website</a>.
+</p>
+
+With postgres running, create databases for development and test using PostgreSQL's `createdb` command:
+
+```shell
+createdb bookshelf_development
+createdb bookshelf_test
+```
+
+In Hanami, [providers](/v2.0/architecture/providers/) offer a mechanism for configuring and using dependencies, like databases, within your application.
+
+Copy and paste the following provider into a new file at `config/providers/persistence.rb`:
+
+```ruby
+# frozen_string_literal: true
+
+Hanami.app.register_provider :persistence, namespace: true do
+  prepare do
+    require "rom-changeset"
+    require "rom/core"
+    require "rom/sql"
+
+    @config = ROM::Configuration.new(:sql, target["settings"].database_url)
+
+    config.plugin(:sql, relations: :auto_restrictions)
+
+    register "config", config
+    register "db", config.gateways[:default].connection
+  end
+
+  start do
+    @config.auto_registration(
+      target.root.join("lib/bookshelf/persistence"),
+      namespace: "Bookshelf::Persistence"
+    )
+
+    register "rom", ROM.container(@config)
+  end
+end
+```
```

Comment:
Feels like we could _probably_ get away without:

`require "rom-changeset"`

`config.plugin(:sql, relations: :auto_restrictions)`

Reply:
Seems we certainly can!